ip-geo.html

```diff
@@ -1,75 +1,93 @@
 
-            // message to display if the client country and state is disabled
-            /* MODIFY THIS TO CHANGE BOT DISABLED DISPLAY TEXT */
-            var disabledMessageText = "The health bot is not currently available in your area. THIS MESSAGE ONLY SHOWS IF DISABLED BY COUNTRY AND STATE!!!";
+        // Track whether or not the bot should be enabled
+        var enableBot = false;
 
-            // define the list of enabled regions to use the healthbot currently
-            /* MODIFY THIS LIST TO ENABLE HEALTHBOT IN SPECIFIED STATES */
-            var enabledCountry = "US";
-            var enabledStates = ['California', 'Washington', 'New York'];
+        // define the list of enabled regions to use the healthbot currently
+        /* MODIFY THIS LIST TO ENABLE HEALTHBOT IN SPECIFIED STATES/REGIONS */
+        var enabledCountry = "US";
+        var enabledStates = [
+        'California', 
+        'New York', 
+        'Washington'
+        ];
 
-            // bool to track whether or not the bot is enabled
-            var enableBot = false;
+        try {
+            // Lowercase all states
+            var lowerCaseEnabledStates = enabledStates.map(function (value) {
+                return value.toLowerCase()
+            });
 
-            try {
-                // lowercase all enbledStates
-                var lowerCaseEnabledStates = enabledStates.map(function (value) {
-                    return value.toLowerCase()
-                });
+            // Is user in permitted state/country?
+            var stateEnabled = lowerCaseEnabledStates.indexOf(json.region.toLowerCase());
+            var countryEnabled = enabledCountry.toLowerCase() === json.country_code.toLowerCase();
 
-                // determine if the current client is in one of the enabled states and country
-                var statedEnabled = lowerCaseEnabledStates.indexOf(json.region.toLowerCase());
-                var countryEnabled = enabledCountry.toLowerCase() === json.country_code.toLowerCase();
+            // Should we show the bot?
+            enableBot = stateEnabled !== -1 && countryEnabled;
+        } catch (error) {
+            console.log(error);
+        }
 
-                // determine if we should show the bot
-                enableBot = statedEnabled === 1 && countryEnabled;
-            } catch (error) {
-                console.log(error);
-            }
 
-            // use the value of enableBot to determine whether or not to display the healthbot
-            if (enableBot) {
-                healthBotEnabled.textContent = "ENABLED";
-            } else {
-                healthBotEnabled.textContent = "DISABLED";
-                var disabledMessage = document.getElementById("disabled_message");
-                disabledMessage.innerHTML = "<p>" + disabledMessageText + "</p>";
-            }
+        /* Now we take action based on enableBot result
```

Comment:
Should we just move this action outside and have this function simply return true/fasle?

Reply:
Probably.  :) 

Lets take this one since it works and improves, and then keep iterating.